rust_src/src/math.rs

```diff
@@ -260,25 +261,30 @@ pub enum ArithComparison {
     GrtrOrEqual,
 }
 
-pub fn arithcompare(obj1: LispObject, obj2: LispObject, comparison: ArithComparison) -> bool {
+pub fn arithcompare(
+    obj1: LispNumberOrFloatOrMarker,
```

Comment:
Maybe make these generic? `impl Into<LispNumberOrFloatOrMarker`, or whatever the syntax is. That would get rid of casting at call sites.

Reply:
Hmm, perhaps. Let me see how complex that is.

Reply:
Since the most common type is LispObject, I moved the more specific type into an internals function.